src/machine-learning/machine-learning.ts

```diff
@@ -229,68 +232,153 @@ export class MachineLearning implements FirebaseServiceInterface {
   }
 }
 
-
 /**
  * A Firebase ML Model output object.
  */
 export class Model {
-  public readonly modelId: string;
-  public readonly displayName: string;
-  public readonly tags?: string[];
-  public readonly createTime: string;
-  public readonly updateTime: string;
-  public readonly validationError?: string;
-  public readonly published: boolean;
-  public readonly etag: string;
-  public readonly modelHash?: string;
-
-  public readonly tfliteModel?: TFLiteModel;
-
-  constructor(model: ModelResponse) {
-    if (!validator.isNonNullObject(model) ||
-      !validator.isNonEmptyString(model.name) ||
-      !validator.isNonEmptyString(model.createTime) ||
-      !validator.isNonEmptyString(model.updateTime) ||
-      !validator.isNonEmptyString(model.displayName) ||
-      !validator.isNonEmptyString(model.etag)) {
-      throw new FirebaseMachineLearningError(
-        'invalid-server-response',
-        `Invalid Model response: ${JSON.stringify(model)}`);
+  private model: ModelResponse;
+  private readonly client?: MachineLearningApiClient;
+
+  constructor(model: ModelResponse, client: MachineLearningApiClient) {
+    Model.validateModel(model);
+    this.model = model;
+    // Remove '@type' field. We don't need it.
+    if ((this.model as {[key: string]: any})["@type"]) {
+      delete (this.model as {[key: string]: any})["@type"];
     }
 
-    this.modelId = extractModelId(model.name);
-    this.displayName = model.displayName;
-    this.tags = model.tags || [];
-    this.createTime = new Date(model.createTime).toUTCString();
-    this.updateTime = new Date(model.updateTime).toUTCString();
-    if (model.state?.validationError?.message) {
-      this.validationError = model.state?.validationError?.message;
+    this.client = client;
+  }
+
+  get modelId(): string {
+    return extractModelId(this.model.name);
+  }
+
+  get displayName(): string {
+    return this.model.displayName!;
+  }
+
+  get tags(): string[] {
+    return this.model.tags || [];
+  }
+
+  get createTime(): string {
+    return new Date(this.model.createTime).toUTCString();
+  }
+
+  get updateTime(): string {
+    return new Date(this.model.updateTime).toUTCString();
+  }
+
+  get validationError(): string | undefined {
+    return this.model.state?.validationError?.message;
+  }
+
+  get published(): boolean {
+    return this.model.state?.published || false;
+  }
+
+  get etag(): string {
+    return this.model.etag;
+  }
+
+  get modelHash(): string | undefined {
+    return this.model.modelHash;
+  }
+
+  get tfliteModel(): TFLiteModel | undefined {
+    if (this.model.tfliteModel) {
+      return {
```

Comment:
If there's a reason why we can't do `return this.model.tfliteModel`, let's add a comment here to indicate why.

Reply:
Done.